### PR TITLE
all: update exp/slices and fix call sites

### PR DESCRIPTION
**Pull request by danderson**
slices.SortFunc suffered a late-in-cycle API breakage.

Updates #cleanup

---

I had to import cmp.Compare into cmpx, since we don't yet want to depend on Go 1.21.

### CODE REVIEW

File: net/dns/manager.go [hunk 1]
```diff
@@ -139,14 +140,14 @@ func compileHostEntries(cfg Config) (hosts []*HostEntry) {
...
-	slices.SortFunc(hosts, func(a, b *HostEntry) bool {
+	slices.SortFunc(hosts, func(a, b *HostEntry) int {
 		if len(a.Hosts) == 0 {
-			return false
+			return -1
 		}
 		if len(b.Hosts) == 0 {
-			return true
+			return 1
 		}
```
Comment (bradfitz) [inline net/dns/manager.go:149]: what if they're both len 0?

File: net/dnsfallback/dnsfallback.go [hunk 2]
```diff
@@ -76,11 +76,27 @@ func MakeLookupFunc(logf logger.Logf, netMon *netmon.Monitor) func(ctx context.C
...
 			// Wait for a response from the main function
 			oldAddrs := <-addrsCh
-			slices.SortFunc(oldAddrs, func(a, b netip.Addr) bool { return a.Less(b) })
+			slices.SortFunc(oldAddrs, func(a, b netip.Addr) int {
+				if a.Less(b) {
+					return -1
+				} else if b.Less(a) {
+					return 1
+				} else {
+					return 0
```
Comment (bradfitz) [inline net/dnsfallback/dnsfallback.go:97]: amusingly we used to have Compare and as part of putting it in std, they asked us to remove it: https://pkg.go.dev/inet.af/netaddr#IP.Compare

Now we have evidence we need it back :)

Comment (bradfitz) [reply net/dnsfallback/dnsfallback.go:97]: Should we add AddrCompare to https://pkg.go.dev/go4.org/netipx ?

File: net/dns/manager.go (hunk 1 above)
Comment (danderson) [reply net/dns/manager.go]: Added a case for that.

File: net/dnsfallback/dnsfallback.go (hunk 2 above)
Comment (bradfitz) [reply net/dnsfallback/dnsfallback.go:97]: https://github.com/go4org/netipx/pull/14

File: cmd/tailscale/cli/exitnode.go [hunk 3]
```diff
@@ -227,19 +228,21 @@ func filterFormatAndSortExitNodes(peers []*ipnstate.PeerStatus, filterBy string)
 // sortPeersByPriority sorts a slice of PeerStatus
 // by location.Priority, in order of highest priority.
 func sortPeersByPriority(peers []*ipnstate.PeerStatus) {
-	slices.SortFunc(peers, func(a, b *ipnstate.PeerStatus) bool { return a.Location.Priority > b.Location.Priority })
+	slices.SortFunc(peers, func(a, b *ipnstate.PeerStatus) int {
+		return cmpx.Compare(b.Location.Priority, a.Location.Priority)
+	})
```
Comment (bradfitz) [inline cmd/tailscale/cli/exitnode.go:233]: unrelated but this should probably be SortStableFunc if there are a bunch of priorities that are the same and the input was already in some reasonable order (like sorted by name)

but can do later

**Review by bradfitz — APPROVED** [review]

File: cmd/tailscale/cli/exitnode.go (hunk 3 above)
Comment (danderson) [reply cmd/tailscale/cli/exitnode.go]: Switched to SortStableFunc.

**Pull request merged**